core/lib/multivm/src/versions/vm_latest/tests/block_tip.rs

```diff
@@ -211,77 +230,105 @@ fn test_dry_run_upper_bound() {
-    });
+    }));
 
-    let long_bytecode = execute_test(L1MessengerTestData {
+    // long bytecode
+    statistics.push(execute_test(L1MessengerTestData {
         // We have to add 48 since a valid bytecode must have an odd number of 32 byte words
         bytecodes: vec![vec![0; MAX_PUBDATA_PER_BLOCK as usize + 48]; 1],
         ..Default::default()
-    });
+    }));
 
-    let lots_of_small_repeated_writes = execute_test(L1MessengerTestData {
+    // lots of small repeated writes
+    statistics.push(execute_test(L1MessengerTestData {
         // In theory each state diff can require only 5 bytes to be published (enum index + 4 bytes for the key)
         state_diffs: generate_state_diffs(true, true, MAX_PUBDATA_PER_BLOCK as usize / 5),
         ..Default::default()
-    });
+    }));
 
-    let lots_of_big_repeated_writes = execute_test(L1MessengerTestData {
+    // lots of big repeated writes
+    statistics.push(execute_test(L1MessengerTestData {
         // Each big write will approximately require 32 bytes to encode
         state_diffs: generate_state_diffs(true, false, MAX_PUBDATA_PER_BLOCK as usize / 32),
         ..Default::default()
-    });
+    }));
 
-    let lots_of_small_initial_writes = execute_test(L1MessengerTestData {
+    // lots of small initial writes
+    statistics.push(execute_test(L1MessengerTestData {
         // Each initial write will take at least 32 bytes for derived key + 5 bytes for value
         state_diffs: generate_state_diffs(false, true, MAX_PUBDATA_PER_BLOCK as usize / 37),
         ..Default::default()
-    });
+    }));
 
-    let lots_of_large_initial_writes = execute_test(L1MessengerTestData {
+    // lots of large initial writes
+    statistics.push(execute_test(L1MessengerTestData {
         // Each big write will take at least 32 bytes for derived key + 32 bytes for value
         state_diffs: generate_state_diffs(false, false, MAX_PUBDATA_PER_BLOCK as usize / 64),
         ..Default::default()
-    });
-
-    let max_used_gas = vec![
-        max_logs,
-        max_messages,
-        long_message,
-        max_bytecodes,
-        long_bytecode,
-        lots_of_small_repeated_writes,
-        lots_of_big_repeated_writes,
-        lots_of_small_initial_writes,
-        lots_of_large_initial_writes,
-    ]
-    .into_iter()
-    .max()
-    .unwrap();
+    }));
 
     // We use 2x overhead for the batch tip compared to the worst estimated scenario.
     assert!(
-        max_used_gas * 2 <= BOOTLOADER_BATCH_TIP_OVERHEAD,
+        statistics.iter().map(|s| s.max_used_gas).max().unwrap() * 2
+            <= BOOTLOADER_BATCH_TIP_OVERHEAD,
         "BOOTLOADER_BATCH_TIP_OVERHEAD is too low"
     );
+
+    assert!(
+        statistics
+            .iter()
+            .map(|s| s.pubdata_published)
+            .max()
+            .unwrap()
+            * 2
+            <= MAX_BATCH_TIP_PUBDATA_PUBLISHED as u32,
+        "MAX_BATCH_TIP_PUBDATA_PUBLISHED is too low"
+    );
```

Comment:
Now you lacking of info about the test by itself. 
You can't distinguish where was a failure

Reply:
Not sure that I understand your point. Do you mean like test case?

Reply:
Yes, I think Danil is talking about providing test case name in panic message

Reply:
Added logging for tests